CMakeLists.txt

```diff
@@ -5,6 +5,9 @@ project(mender)
 # Optimize aggressively for size when building MinSizeRel
 string(APPEND CMAKE_CXX_FLAGS_MINSIZEREL " -Oz")
 
+# Enable Link-Time-Optimizations when building MinSizeRel
+string(APPEND CMAKE_CXX_FLAGS_MINSIZEREL " -flto")
```

Comment:
Not sure how portable this is. AFAIK, only recent versions of GCC and Clang can do this (reliably).

Reply:
Yes, I too am a little unsure about it. I'll check if we can do a check with CMake, and then add it conditionally.

Reply:
Updated

Reply:
Nice!